images/ansible-operator/Dockerfile

```diff
@@ -11,20 +11,22 @@ RUN rustc --version
 
 # Copy python dependencies (including ansible) to be installed using Pipenv
 COPY images/ansible-operator/Pipfile* ./
+
 # Instruct pip(env) not to keep a cache of installed packages,
 # to install into the global site-packages and
 # to clear the pipenv cache as well
-ENV PIP_NO_CACHE_DIR=1 \
-    PIPENV_SYSTEM=1 \
-    PIPENV_CLEAR=1
+ENV PIP_NO_CACHE_DIR=1 
+
 # Ensure fresh metadata rather than cached metadata, install system and pip python deps,
 # and remove those not needed at runtime.
 RUN set -e && yum clean all && rm -rf /var/cache/yum/* \
   && yum update -y \
-  && yum install -y libffi-devel openssl-devel python39-devel gcc python39-pip python39-setuptools \
-  && pip3 install --upgrade pip~=23.3.2 \
-  && pip3 install pipenv==2023.11.15 \
-  && pipenv install --deploy \
+  && yum install -y python3.11 \
+  && yum install -y libffi-devel openssl-devel gcc python3.11-devel python3.11-pip python3.11-setuptools \
+  && pip3.11 install --upgrade pip~=24.2 \
+  && pip3.11 install pipenv \
+  && pipenv requirements > requirements.txt \
```

Comment:
Using pipenv for now to reuse piplock.

Reply:
I think creating the requirements.txt file while building the image may have some issues while reproducing it locally as was the ask in https://github.com/operator-framework/operator-sdk/issues/4237.
We can have fixed dependencies list beforehand and then use it while building the image. This can be achieved in 2 ways:
- Either we continue with the same method and generate the Pipfile.lock as mentioned here: https://github.com/operator-framework/ansible-operator-plugins/blob/42b5d80c75f1ddda8f2dbe1629b9454d366a8d6a/images/ansible-operator/README.md?plain=1#L1-L7
- Or we generate the requirements.txt file along with Pipfile.lock and then use the requirements.txt file during building the image.

Reply:
> * Or we generate the requirements.txt file along with Pipfile.lock and then use the requirements.txt file during building the image.
The requirements.txt is generated based on Pipfile.lock, refer https://pipenv.pypa.io/en/latest/cli.html#requirements.
Regarding the issue mentioned, the dependencies are managed via pipfile.

Reply:
Ack. But I think the Pipfile.lock file is not updated after your last change to the Pipfile. I used both the files to generate the requirements.txt.backup file. After that I updated the Pipfile.lock using the latest Pipfile and then generated the requirements.txt. Following is the diff between the files:
```shell
# diff -y requirements.txt requirements.txt.backup 
-i https://pypi.org/simple					-i https://pypi.org/simple
annotated-types==0.7.0; python_version >= '3.8'			annotated-types==0.7.0; python_version >= '3.8'
							      >	ansible==10.4.0; python_version >= '3.10'
ansible-core==2.17.4; python_version >= '3.10'			ansible-core==2.17.4; python_version >= '3.10'
ansible-runner==2.4.0; python_version >= '3.9'			ansible-runner==2.4.0; python_version >= '3.9'
ansible-runner-http==1.0.0					ansible-runner-http==1.0.0
authlib==1.3.2; python_version >= '3.8'				authlib==1.3.2; python_version >= '3.8'
cachetools==5.5.0; python_version >= '3.7'			cachetools==5.5.0; python_version >= '3.7'
certifi==2024.8.30; python_version >= '3.6'			certifi==2024.8.30; python_version >= '3.6'
cffi==1.17.1; platform_python_implementation != 'PyPy'		cffi==1.17.1; platform_python_implementation != 'PyPy'
charset-normalizer==3.3.2; python_full_version >= '3.7.0'	charset-normalizer==3.3.2; python_full_version >= '3.7.0'
click==8.1.7; python_version >= '3.7'				click==8.1.7; python_version >= '3.7'
cryptography==43.0.1; python_version >= '3.7'			cryptography==43.0.1; python_version >= '3.7'
docutils==0.21.2; python_version >= '3.9'			docutils==0.21.2; python_version >= '3.9'
dparse==0.6.4b0; python_version >= '3.7'			dparse==0.6.4b0; python_version >= '3.7'
filelock==3.12.4; python_version >= '3.8'			filelock==3.12.4; python_version >= '3.8'
google-auth==2.35.0; python_version >= '3.7'		      |	google-auth==2.34.0; python_version >= '3.7'
idna==3.10; python_version >= '3.6'			      |	idna==3.8; python_version >= '3.6'
jinja2==3.1.4; python_version >= '3.7'				jinja2==3.1.4; python_version >= '3.7'
kubernetes==29.0.0; python_version >= '3.6'			kubernetes==29.0.0; python_version >= '3.6'
lockfile==0.12.2						lockfile==0.12.2
markdown-it-py==3.0.0; python_version >= '3.8'			markdown-it-py==3.0.0; python_version >= '3.8'
markupsafe==2.1.5; python_version >= '3.7'			markupsafe==2.1.5; python_version >= '3.7'
marshmallow==3.22.0; python_version >= '3.8'			marshmallow==3.22.0; python_version >= '3.8'
mdurl==0.1.2; python_version >= '3.7'				mdurl==0.1.2; python_version >= '3.7'
oauthlib==3.2.2; python_version >= '3.6'			oauthlib==3.2.2; python_version >= '3.6'
packaging==24.1; python_version >= '3.8'			packaging==24.1; python_version >= '3.8'
pexpect==4.9.0							pexpect==4.9.0
psutil==6.0.0; python_version >= '2.7' and python_version not	psutil==6.0.0; python_version >= '2.7' and python_version not
ptyprocess==0.7.0						ptyprocess==0.7.0
pyasn1==0.6.1; python_version >= '3.8'				pyasn1==0.6.1; python_version >= '3.8'
pyasn1-modules==0.4.1; python_version >= '3.8'			pyasn1-modules==0.4.1; python_version >= '3.8'
pycparser==2.22; python_version >= '3.8'			pycparser==2.22; python_version >= '3.8'
pydantic==2.9.2; python_version >= '3.8'		      |	pydantic==2.9.1; python_version >= '3.8'
pydantic-core==2.23.4; python_version >= '3.8'		      |	pydantic-core==2.23.3; python_version >= '3.8'
pygments==2.18.0; python_version >= '3.8'			pygments==2.18.0; python_version >= '3.8'
python-daemon==3.0.1; python_version >= '3'			python-daemon==3.0.1; python_version >= '3'
python-dateutil==2.9.0.post0; python_version >= '2.7' and pyt	python-dateutil==2.9.0.post0; python_version >= '2.7' and pyt
pyyaml==6.0.2; python_version >= '3.8'				pyyaml==6.0.2; python_version >= '3.8'
requests==2.31.0; python_version >= '3.7'			requests==2.31.0; python_version >= '3.7'
requests-oauthlib==2.0.0; python_version >= '3.4'		requests-oauthlib==2.0.0; python_version >= '3.4'
requests-unixsocket==0.3.0					requests-unixsocket==0.3.0
resolvelib==1.0.1						resolvelib==1.0.1
rich==13.8.1; python_full_version >= '3.7.0'			rich==13.8.1; python_full_version >= '3.7.0'
rsa==4.9; python_version >= '3.6' and python_version < '4'	rsa==4.9; python_version >= '3.6' and python_version < '4'
ruamel.yaml==0.18.6; python_version >= '3.7'			ruamel.yaml==0.18.6; python_version >= '3.7'
ruamel.yaml.clib==0.2.8; python_version < '3.13' and platform	ruamel.yaml.clib==0.2.8; python_version < '3.13' and platform
safety==3.2.7; python_version >= '3.7'				safety==3.2.7; python_version >= '3.7'
safety-schemas==0.0.5; python_version >= '3.7'			safety-schemas==0.0.5; python_version >= '3.7'
setuptools==75.1.0; python_version >= '3.8'		      |	setuptools==74.1.2; python_version >= '3.8'
shellingham==1.5.4; python_version >= '3.7'			shellingham==1.5.4; python_version >= '3.7'
six==1.16.0; python_version >= '2.7' and python_version not i	six==1.16.0; python_version >= '2.7' and python_version not i
typer==0.12.5; python_version >= '3.7'				typer==0.12.5; python_version >= '3.7'
typing-extensions==4.12.2; python_version >= '3.8'		typing-extensions==4.12.2; python_version >= '3.8'
urllib3==1.26.20; python_version >= '2.7' and python_version 	urllib3==1.26.20; python_version >= '2.7' and python_version 
websocket-client==1.8.0; python_version >= '3.8'		websocket-client==1.8.0; python_version >= '3.8'

```

Reply:
Ah, good catch